lighthouse-core/test/gather/gatherers/seo/font-size-test.js

```diff
@@ -13,47 +13,83 @@ let fontSizeGather;
 const smallText = ' body small text ';
 const bigText = 'body big text';
 const failingText = 'failing text';
-const bodyNode = {nodeId: 3, nodeName: 'BODY', parentId: 1};
-const failingNode = {nodeId: 10, nodeName: 'P', parentId: 3};
+const bodyNode = {nodeId: 3, backendNodeId: 102, nodeName: 'BODY', parentId: 1, fontSize: '10px'};
+const failingNode = {nodeId: 10, backendNodeId: 109, nodeName: 'P', parentId: 3};
 const nodes = [
-  {nodeId: 1, nodeName: 'HTML'},
-  {nodeId: 2, nodeName: 'HEAD', parentId: 1},
+  {nodeId: 1, backendNodeId: 100, nodeName: 'HTML'},
+  {nodeId: 2, backendNodeId: 101, nodeName: 'HEAD', parentId: 1},
   bodyNode,
   {
     nodeId: 4,
+    backendNodeId: 103,
     nodeValue: 'head text',
     nodeType: FontSizeGather.TEXT_NODE_TYPE,
     parentId: 2,
   },
   {
     nodeId: 5,
+    backendNodeId: 104,
     nodeValue: smallText,
     nodeType: FontSizeGather.TEXT_NODE_TYPE,
     parentId: 3,
   },
-  {nodeId: 6, nodeName: 'H1', parentId: 3},
+  {nodeId: 6, backendNodeId: 105, nodeName: 'H1', parentId: 3},
   {
     nodeId: 7,
+    backendNodeId: 106,
     nodeValue: bigText,
     nodeType: FontSizeGather.TEXT_NODE_TYPE,
     parentId: 6,
   },
-  {nodeId: 8, nodeName: 'SCRIPT', parentId: 3},
+  {nodeId: 8, backendNodeId: 107, nodeName: 'SCRIPT', parentId: 3},
   {
     nodeId: 9,
+    backendNodeId: 108,
     nodeValue: 'script text',
     nodeType: FontSizeGather.TEXT_NODE_TYPE,
     parentId: 8,
   },
   failingNode,
   {
     nodeId: 11,
+    backendNodeId: 110,
     nodeValue: failingText,
     nodeType: FontSizeGather.TEXT_NODE_TYPE,
     parentId: 10,
   },
 ];
 
+const stringsMap = {};
+const strings = [];
+const dedupe = (value) => {
+  if (stringsMap[value]) {
+    return stringsMap[value];
+  }
+
+  const index = strings.length;
+  stringsMap[value] = index;
+  strings.push(value);
+  return index;
+};
+const snapshot = {
+  documents: [
+    {
+      nodes: {
+        nodeType: nodes.map(node => node.nodeType),
+        nodeName: nodes.map(node => dedupe(node.nodeName)),
+        nodeValue: nodes.map(node => dedupe(node.nodeValue)),
+        backendNodeId: nodes.map(node => node.backendNodeId),
+        parentIndex: nodes.map(node => nodes.findIndex(pNode => node.parentId === pNode.nodeId)),
+      },
+      layout: {
+        nodeIndex: nodes.map((_, i) => i), // this isn't very accurate
```

Comment:
mind elaborating on the comment a bit? :)

is it inaccurate because the nodeIndexes are usually very large? are they usually not sequential? do any of these things affect the fidelity of the tests?

Reply:
If you take a look at an actual snapshot, you'll see that there isn't a 1:1 mapping between Nodes and style entries. That's because some elements (like text nodes) don't have styles. Here, I'm being lazy and saying that all nodes have computed styles, and that the mapping is just the identity function.

It is indeed missing out on a small amount of functionality, namely the nodeId => styleIndex mapping in the implementation. it wasn't immediately clear how to generate that part of the fake snapshot so I just punted it. I could look into it a bit more.

Reply:
let's trust the protocol documentation

https://chromedevtools.github.io/devtools-protocol/tot/DOMSnapshot#method-captureSnapshot
> The DOMNode at index 0 corresponds to the root document.

If it's wrong then testing will catch it :)